### ISSUE TITLE
`parent_config` config does not include `custom_rules`

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3468]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I'm trying out the new `parent_config` property, but it seems it does not include custom rules.

I noticed since I have a lot of `// swiftlint:disable:next final_class` for [this rule](https://github.com/sindresorhus/swiftlint-config/blob/723a1fa4dd4af7dd0e0d87197997239dea9a0a1b/.swiftlint.yml#L187-L189) and it started complaining about "Superfluous Disable Command Violation: 'final_class' is not a valid SwiftLint rule. Please remove it from the disable command. (superfluous_disable_command)".

##### Complete output when running SwiftLint, including the stack trace and command used

N/A

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.42.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? GitHub Releases
* Paste your configuration file:

```yml
parent_config: https://github.com/sindresorhus/swiftlint-config/raw/723a1fa/.swiftlint.yml
```

---

// @fredpi

**Comment by fredpi** [comment]
@sindresorhus Thanks for reporting! Will look into this tomorrow

**Comment by fredpi** [comment]
@sindresorhus At first, I thought this issue is related to https://github.com/realm/SwiftLint/issues/3472, but now I believe it's a different issue. I will investigate further tomorrow.

**Comment by fredpi** [comment]
I figured out the root cause of the issue (which turns out to be only one of multiple issues with `custom_rules`merging), but fixing it properly will require some time.

**Comment by fredpi** [comment]
@sindresorhus Okay, it took less time than expected. See https://github.com/realm/SwiftLint/pull/3473. Would you mind checking if that PR fixes the issue for you?

**Comment by mgray88** [comment]
Would that PR also fix overriding a custom rule of the same name from a parent config? If I just use the name of the custom rule to configure the severity it has no effect. If I redefine it under `custom_rules` it runs both checks and outputs a warning as well as an error

**Comment by fredpi** [comment]
@mgray88 That's a separate issue, but I updated the PR to also address it. In addition, I opened #3477 to track that bug separately.

**Comment by fredpi** [comment]
Closed via https://github.com/realm/SwiftLint/pull/3473.

**Comment by jpsim** [comment]
UnfortunatelyI had to revert #3473 in #3503.

**Comment by jpsim** [comment]
Fixed again in https://github.com/realm/SwiftLint/pull/3538
